aten/src/ATen/test/tensor_iterator_test.cpp

```diff
@@ -89,9 +89,27 @@ TEST(TensorIteratorTest, SerialLoopPointwise_##name) {
   ASSERT_ANY_THROW(out.equal(expected));                                                            \
 }
 
+#define COMPARISON_TEST_ITER_FOR_TYPE(ctype,name)                                          \
+TEST(TensorIteratorTest, ComparisonLoopBinary_##name) {                                    \
+  auto in1 = random_tensor_for_type(k##name);                                              \
+  auto in2 = random_tensor_for_type(k##name);                                              \
+  Tensor out = at::empty({0}, in1.options().dtype(kBool));                                 \
+  Tensor diff;                                                                             \
+  if (k##name == kByte) {                                                                  \
+    diff = in2.to(kInt).sub(in1.to(kInt));                                                 \
```

Comment:
Why is this necessary? Do we need this for `bool` as well?

Reply:
Yes we need. I also added test for Bool. We need this for unsigned type.

Reply:
Why do we need this? Is it to prevent against integer underflow? It's not completely obvious to a code reader (me) why this needs to be special cased.

Reply:
Added comments

Reply:
Ah, I see now. Thanks for the comment.

`max(a, b) == a` works as well, and might be a little more elegant. The thing I'm worried about is that someone will come along and add a new unsigned type and forget to update this test. But what you have here should be fine.

Reply:
I think it will not work when a == b, then we should return False